shared-module/displayio/Shape.c

```diff
@@ -66,8 +72,58 @@ void common_hal_displayio_shape_set_boundary(displayio_shape_t *self, uint16_t y
     if (self->mirror_x && (start_x > half_width || end_x > half_width)) {
         mp_raise_ValueError_varg(translate("Maximum x value when mirrored is %d"), half_width);
     }
+
+    uint16_t lower_x, upper_x;
+
+    // find x-boundaries for updating based on current data and start_x, end_x
+    if (start_x < self->data[2 * y]) {
+        lower_x = start_x;
+    } else {
+        lower_x = self->data[2 * y];
+    }
```

Comment:
Mind rewriting this bit with MIN and MAX macros from `py/misc.h`? I think it'll make the code much clearer.

Reply:
The code was even crazier when I didn't use these intermediate `upper_x` and `lower_x` variables!

Thanks for pointing out the availability of MAX/MIN in `py.misc.h`.  I incorporated these changes.